R/get_meetnetten_locations.R

```diff
@@ -0,0 +1,178 @@
+#' occurs. For more common species a sample of locations is drawn and the
+#' the selected locations are included in the monitoring scheme. In some cases,
+#' the monitoring project in \href{www.meetnetten.be}{Meetnetten} also contains
+#' locations that are not part of the sample. These locations can be counted
+#' optionally and are indicated by (\code{is_sample} = \code{FALSE}).
+#'
+#' It also occurs that a location becomes inaccessible or that the target
+#' species disappears. Then, a locations can be made inactive (\code{is_active}
+#' = \code{FALSE}), which means that no observations can be recorded any more.
+#'
+#' @importFrom assertthat assert_that
+#' @importFrom dplyr filter arrange collect tbl sql mutate select
+#' @importFrom stringr str_to_lower
+#' @importFrom sf st_as_sf st_drop_geometry
+#'
+#' @export
+#' @family meetnetten
+#' @examples
+#' \dontrun{
+#' library(inbodb)
+#' con <- connect_inbo_dbase("S0008_00_Meetnetten")
+#'
+#' # get locations for a specific monitoring scheme
+#' locations_heivlinder <- get_meetnetten_locations(con,
+#'                                                  scheme_name = "Heivlinder")
+#'
+#' main_locations <- locations_heivlinder$main_locations
+#' sublocations <- locations_heivlinder$sublocations
+#'
+#' # get locations for a specific species_group
+#' locations_dragonflies <- get_meetnetten_locations(con,
+#'                                                   species_group = "libellen")
+#'
+#' main_locations <- locations_dragonflies$main_locations
+#' sublocations <- locations_dragonflies$sublocations
+#'
+#' # Close the connection when done
+#' dbDisconnect(con)
+#' rm(con)
+#' }
+
+get_meetnetten_locations <- function(connection,
+                                    scheme_name = NULL,
+                                    species_group = NULL) {
+
+  species_group_selected <- species_group
+
+  assert_that(inherits(connection, what = "Microsoft SQL Server"),
+              msg = "Not a connection object to database.")
+
+  if (!is.null(scheme_name)) {
+    assert_that(is.character(scheme_name))
+    scheme_name <- str_to_lower(scheme_name)
+  }
+
+  if (!is.null(species_group_selected)) {
+    assert_that(is.character(species_group_selected))
+    species_group_selected <- str_to_lower(species_group_selected)
+  }
+
+  sql_statement <- "SELECT
+    PG.name as species_group,
+    P.name as scheme,
+    L.name as location,
+    L.id,
+    L.parent_id,
+    PL.is_sample,
+    PL.is_active,
+    L.geom.STAsText() as geom
+    FROM staging_meetnetten.locations_location L
+    INNER JOIN staging_meetnetten.projects_projectlocation PL ON PL.location_id = L.id
+    INNER JOIN staging_meetnetten.projects_project P ON P.id = PL.project_id
+    INNER JOIN staging_meetnetten.projects_projectgroup PG on PG.id = P.group_id"
+
+  query_result <- tbl(connection, sql(sql_statement))
+
+  if (!is.null(scheme_name) & !is.null(species_group_selected)) {
```

Comment:
Always use '&&' in if statements
```suggestion
  if (!is.null(scheme_name) && !is.null(species_group_selected)) {
```